demonstrations/tutorial_lcu_blockencoding.py

```diff
@@ -0,0 +1,288 @@
+#
+# If you're up for it, it's illuminating to go through the math and show how :math:`A` comes out on the right
+# side of the equation.
+# (Tip: calculate the action of :math:`\text{PREP}^\dagger` on :math:`\langle 0|`, not on the output
+# state after :math:`\text{SEL} \cdot \text{PREP}`).
+#
+# Otherwise, the intuitive way to understand this equation is that we apply PREP, SEL, and then invert PREP. If
+# we measure :math:`|0\rangle` in the auxiliary qubits, the input state :math:`|\psi\rangle` will be transformed by
+# :math:`A` (up to normalization). The figure below shows this as a circuit with four unitaries in SEL.
+#
+# |
+#
+# .. figure:: ../demonstrations/lcu_blockencoding/schematic.png
+#     :align: center
+#     :width: 50%
+#     :target: javascript:void(0)
+#
+# |
+#
+# The circuit
+#
+# .. math:: U = \text{PREP}^\dagger \cdot \text{SEL} \cdot \text{PREP}
+#
+# is a **block encoding** of :math:`A`, up to normalization. The reason for this name is that if we write :math:`U`
+# as a matrix, the operator :math:`A` is encoded inside a block of :math:`U` as
+#
+# .. math:: U = \begin{bmatrix} A & \cdot \\ \cdot & \cdot \end{bmatrix}.
+#
+# This block is defined by the subspace of all states where the auxiliary qubits are in state
+# :math:`|0\rangle`.
+#
+#
+# PennyLane supports the direct implementation of `prepare <https://docs.pennylane.ai/en/stable/code/api/pennylane.StatePrep.html>`_
+# and `select <https://docs.pennylane.ai/en/stable/code/api/pennylane.Select.html>`_
+# operators. We'll go through them individually and use them to construct a block encoding circuit.
+# Prepare circuits can be constructed using the :class:`~.pennylane.StatePrep` operation, which takes
+# the normalized target state as input:
+
+dev1 = qml.device("default.qubit", wires=1)
+
+# normalized square roots of coefficients
+alphas = (np.sqrt(LCU.coeffs) / np.linalg.norm(np.sqrt(LCU.coeffs)))
+
+
+@qml.qnode(dev1)
+def prep_circuit():
+    qml.StatePrep(alphas, wires=0)
+    return qml.state()
+
+
+print("Target state: ", alphas)
+print("Output state: ", np.real(prep_circuit()))
+
+##############################################################################
+# Similarly, select circuits can be implemented using :class:`~.pennylane.Select`, which takes the
+# target unitaries as input. We specify the control wires directly, but the system wires are inherited
+# from the unitaries. Since :func:`~.pennylane.pauli_decompose` uses a canonical wire ordering, we
+# first map the wires to those used for the system register in our circuit:
+
+dev2 = qml.device("default.qubit", wires=3)
+
+# unitaries
+ops = LCU.ops
+# relabeling wires: 0 → 1, and 1 → 2
+unitaries = [qml.map_wires(op, {0: 1, 1: 2}) for op in ops]
+
+
+@qml.qnode(dev2)
+def sel_circuit(qubit_value):
+    qml.BasisState(qubit_value, wires=0)
+    qml.Select(unitaries, control=0)
+    return qml.expval(qml.PauliZ(2))
+
+qml.draw_mpl(sel_circuit, style='pennylane')([0])
```

Comment:
```suggestion
qml.draw_mpl(sel_circuit, style='pennylane')([0])
plt.show()
```

This will remove the unwanted output from the cell